en/docs/deploy/deploy-api/exposing-apis-via-custom-hostnames.md

```diff
@@ -0,0 +1,191 @@
+            Follow [Publish through Multiple API Gateways]({{base_path}}/deploy/deploy-api/publish-through-multiple-api-gateways/) to understand how to work with multiple Gateway environments.
+
+     3. Optionally, click **Advanced Settings** to add a HTTP(S) context and custom ports.
+
+        Let's add `gateway` as the HTTP(S) context for the `us-region` Gateway environment.
+
+        Add another virtual host `foods.com` by clicking **New VHost** and click **Save** to save the environment.
+
+#### Create a Gateway environment using the configuration file
+
+Follow the instructions below to use the `deployment.toml` file, which is the central configuration file, to configure a Gateway environment that consists of virtual hosts:
+
+1. Open to the `<APIM-HOME>/repository/conf/deployment.toml` file.
+
+2. Create a Gateway environment.
+
+     Create a Gateway environment with the following Gateway configurations, which include `us.wso2.com` and `foods.com` as the custom VHosts and `food` as the custom context.
+
+    !!! note
+
+        - When the WSO2 API Manager server is running, the Gateway environments, which you added via the `deployment.toml` file, are displayed in the Gateway environments list page in the Admin Portal in read-only mode. 
+        - Therefore, ensure to add the VHosts that correspond to the Gateway at the time of adding the Gateway environment itself.
+        - If a VHost is not defined, the default VHost details are assigned to the Gateway environment.
+        - It is not mandatory to specify a context for the VHost.
+
+    ```toml
+    [[apim.gateway.environment]]
+    name = "us-region"
+    display_name = "US Region"
+    type = "hybrid"
+    display_in_api_console = true
+    description = "Gateway environment deployed in the US region."
+    show_as_token_endpoint_url = true
+    service_url = "https://localhost:${mgt.transport.https.port}/services/"
+    username= "${admin.username}"
+    password= "${admin.password}"
+    ws_endpoint = "ws://localhost:9099"
+    wss_endpoint = "wss://localhost:8099"
+    http_endpoint = "http://localhost:${http.nio.port}"
+    https_endpoint = "https://localhost:${https.nio.port}"
+
+    [[apim.gateway.environment.virtual_host]]
+    ws_endpoint = "ws://us.wso2.com:9099"
+    wss_endpoint = "wss://us.wso2.com:8099"
+    http_endpoint = "http://us.wso2.com/gateway"
+    https_endpoint = "https://us.wso2.com/gateway"
+
+    [[apim.gateway.environment.virtual_host]]
+    ws_endpoint = "ws://foods.com:9099"
+    wss_endpoint = "wss://foods.com:8099"
+    http_endpoint = "http://foods.com"
+    https_endpoint = "https://foods.com"
+    ```
+
+3. [Start WSO2 API Manager]({{base_path}}/install-and-setup/install/running-the-product/#starting-the-server).
+
+      This will start WSO2 API Manager in the all-in-one mode, which includes the default Gateway as well.
+
+### Step 2 - Start the second Gateway
+
+[Add Content]
```

Comment:
@renuka-fernando , please add content.

Reply:
added content to start the second server with port offset and assigning gateway labels